Automattic/wp-calypso
# Gutenboarding: Integrate PlansTable & PlansDetails into PlansGrid.

#### Changes proposed in this Pull Request

- move `onboarding-block/plans` into `components/plans/plans-table` & `components/plans/plans-details`.
- insert `plans-table` & `plans-details` into `plans-grid`.
- remove title & subtitle from the mockup
- wire-up plan selection logic

#### Testing instructions
Note: works behind feature-flag atm.

* Go to `/new?flags=gutenboarding/plans-grid`
* Press the Plan button (top-right) which should have the value _Free Plan_ with an empty state.
* Select another plan and click Confirm button.
* The Plan button should now display the new selected plan.
* When returning later to Gutenboarding the selected (and confirmed) plan label should be still displayed in the Plan button and pre-selected when opening the Plans grid.
* After completing Gutenboarding with site creation, selected plan should be reset.

#### Screenshot

![image](https://user-images.githubusercontent.com/1287077/81296945-ef14a300-9072-11ea-8671-4ba99b0e9b63.png)

Fixes part of https://github.com/Automattic/wp-calypso/issues/41787
